src/components/views/dialogs/BaseDialog.js

```diff
@@ -49,6 +52,9 @@ export default React.createClass({
     },
 
     _onKeyDown: function(e) {
+        if (this.props.onKeyDown) {
+            this.props.onKeyDown(e);
```

Comment:
We should probably return here, right? Otherwise we call onKeyDown and potentially also onFinished and onEnterPressed?

Reply:
I was debating what to do here and eventually decided that if you supplied both callbacks, the least surprising behaviour would be for them to both be called, although I appreciate any behaviour here is a bit odd and you probably never want to supply both.

Reply:
Then I guess this is probably fine. I think it'd be quite obviously weird if someone wrote something that provided both props.